<?xml version="1.0"?>
<launch>

  <arg name="force_or_potential" default="false" /> 


  <node name="slave_controller" pkg="haptic_teleoperation" type="controller_ros"   ns="/haptic_teleoperation">
    <rosparam command="load" file="$(find haptic_teleoperation)/config/slave_controller_params_ground.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/slave_params.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/master_params.yaml" />
    <remap from="/ground_truth/state" to="/RosAria/pose"/> 
    <remap from="/cmd_vel" to="/RosAria/cmd_vel"/>
  </node>

  <node name="master_controller" pkg="haptic_teleoperation" type="controller_ros"  ns="/haptic_teleoperation">
    <rosparam command="load" file="$(find haptic_teleoperation)/config/slave_params.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/master_controller_params_ground.yaml" />
    <rosparam command="load" file="$(find haptic_teleoperation)/config/master_params.yaml" />
    <remap from="/ground_truth/state" to="/RosAria/pose"/> 
    <remap from="/cmd_vel" to="/RosAria/cmd_vel"/>
  </node>

 <!-- Convert the laser data into point clout data (subscribe for lasersacn and publish point cloud)-->
<!-- <node name="laser2pointcloud" pkg="haptic_teleoperation" type="laser2pointcloud" ns="/haptic_teleoperation" output="screen">
	<remap from="/laserscan" to="/scan"/>
  </node> -->

<group if="$(arg force_or_potential)">
<node name="potential_field" pkg="haptic_teleoperation" type="potential_field" ns="/haptic_teleoperation" >
    <rosparam command="load" file="$(find haptic_teleoperation)/config/potential_field.yaml" />
    <remap from="/ground_truth/state" to="/RosAria/pose"/> 
    <remap from="/cmd_vel" to="/RosAria/cmd_vel"/>
  </node>
 </group>

<group unless="$(arg force_or_potential)">
 <node name="force_field" pkg="haptic_teleoperation" type="force_field" output="screen"  ns="/haptic_teleoperation">
    <rosparam command="load" file="$(find haptic_teleoperation)/config/force_field_params.yaml" />
    <remap from="/ground_truth/state" to="/RosAria/pose"/> 
    <remap from="/cmd_vel" to="/RosAria/cmd_vel"/>
  </node>
</group>

 <!-- <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen"/> --> 
  <!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(find haptic_teleoperation)/rviz/config.rviz" required="true" /> -->
</launch>
